### PR TITLE
Fix front2

### PR DESCRIPTION
**Pull request by stepanenkoales**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,18 +3,67 @@ import { HomePage } from './pages/homePage'
...
 import { ResetForm } from './pages/resetForm'
-
+import { AdminForm } from './pages/adminForm'
+import { AddCategoryForm } from './pages/addCategoryForm'
+import { AddItemForm } from './pages/addItemForm'
+import { UpdateItemForm } from './pages/updateItemForm'
+import { DeleteItemForm } from './pages/deleteItemForm'
 import { Routes, Route } from 'react-router-dom'
 import { routes } from './config/routes'
+import { RequireAuth } from './hoc/requireAuth'
```
Comment (tresdosdos) [inline src/App.js]: could you rename it to AdminGuard?

File: src/hoc/requireAuth.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { AuthContext } from './authProvider'
+import { storageService } from '../utils/storage.service'
+import { httpsService } from '../utils/https.service'
+import { routes } from '../config/routes'
+
+export const RequireAuth = ({ children }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const { user, setUser } = useContext(AuthContext)
+  const location = useLocation()
+  const accessToken = storageService.get('accessToken')
```
Comment (tresdosdos) [inline src/hoc/requireAuth.js]: move this into useEffect because you don't use it right in component

File: src/pages/addCategoryForm.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import { routes } from '../config/routes'
+import { Form, Input, Button, Space, Select } from 'antd'
+import { MinusCircleOutlined, PlusOutlined } from '@ant-design/icons'
+import { notificationService } from '../utils/notification.service'
+import { httpsService } from '../utils/https.service'
+import '../styles/adminForm.scss'
+
+export const AddCategoryForm = () => {
+  const [categories, setCategories] = useState([])
+  const { Option } = Select
```
Comment (tresdosdos) [inline src/pages/addCategoryForm.js]: you should extract this from the component

File: src/pages/deleteItemForm.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import { notificationService } from '../utils/notification.service'
+import { Form, Button, Select } from 'antd'
+
+import '../styles/adminForm.scss'
+
+export const DeleteItemForm = () => {
+  const [categories, setCategories] = useState([])
+  const [items, setItems] = useState([])
+
+  const { Option } = Select
```
Comment (tresdosdos) [inline src/pages/deleteItemForm.js]: extract from the component

File: src/pages/homePage.js [hunk 5]
```diff
@@ -1,34 +1,173 @@
...
+  const [categories, setCategories] = useState([])
+  const [subCategories, setSubCategories] = useState([])
+  const [categoryId, setCategoryId] = useState(null)
+  const [searchValue, setSearchValue] = useState(null)
+  const [totalPages, setTotalPages] = useState(0)
+
+  const { Header, Content, Footer, Sider } = Layout
+  const { Search } = Input
+
+  const parentCategories = categories.filter((item) => item.parentId === null)
```
Comment (tresdosdos) [inline src/pages/homePage.js]: wrap it with useMemo hook

File: src/pages/homePage.js [hunk 6]
```diff
@@ -1,34 +1,173 @@
...
+      })
+  }, [])
+
+  const handleSubCategoriesChange = (e) => {
+    const chosenSubCategories = categories.filter(
+      (category) => category.parentId === e.key
+    )
+    setSubCategories(chosenSubCategories)
+  }
+
```
Comment (tresdosdos) [inline src/pages/homePage.js]: wrap it with useCallback

File: src/pages/homePage.js [hunk 7]
```diff
@@ -1,34 +1,173 @@
...
+    {
+      title: 'Price',
+      dataIndex: 'price',
+      sorter: (a, b) => a.price - b.price,
+    },
+    {
+      title: 'Description',
+      dataIndex: 'description',
+    },
+  ]
```
Comment (tresdosdos) [inline src/pages/homePage.js]: extract this from the component

File: src/pages/homePage.js [hunk 8]
```diff
@@ -1,34 +1,173 @@
...
+            {subCategories.map((item) => (
+              <Menu.Item key={item.id}>{item.category}</Menu.Item>
+            ))}
+          </Menu>
+        </Sider>
+
+        <div className="content">
+          <Content className="site-layout">
+            <Table
+              rowKey={(items) => items.id}
```
Comment (tresdosdos) [inline src/pages/homePage.js]: extract this fn from the component

File: src/styles/homeForm.scss [hunk 9]
```diff
@@ -10,23 +10,55 @@
...
+.ant-row {
+    flex-flow: nowrap !important;
+}
+
+.ant-input {
+    width: fit-content !important;
+}
+
+.ant-table-pagination-right {
+    justify-content: center !important;
```
Comment (tresdosdos) [inline src/styles/homeForm.scss:46]: can we do this without important statements?

**Review by tresdosdos — CHANGES_REQUESTED** [review]

File: src/styles/homeForm.scss (hunk 9 above)
Comment (stepanenkoales) [reply src/styles/homeForm.scss:46]: no, I tried

File: src/pages/homePage.js [hunk 10]
```diff
@@ -1,34 +1,182 @@
...
+
+const columns = [
+  {
+    title: 'image',
+    dataIndex: 'image',
+  },
+  {
+    title: 'Name',
+    dataIndex: 'name',
+    sorter: (a, b) => a.name - b.name,
```
Comment (tresdosdos) [inline src/pages/homePage.js:20]: please use a.name.localCompare(b.name) for string. don't use math operands

File: src/pages/homePage.js [hunk 11]
```diff
@@ -1,34 +1,182 @@
...
+    })
+    const { rows } = response
+    setItems(rows)
+  }
+
+  const onSearch = async (value) => {
+    if (!value.trim()) {
+      return null
+    }
+    setSearchValue(value.trim())
```
Comment (tresdosdos) [inline src/pages/homePage.js:102]: do you really need to change state with trim?

**Review by tresdosdos — APPROVED** [review]

**Pull request merged**